From: Cyril Brulebois <kibi@debian.org>
Date: Mon, 06 Feb 2023 04:06:22 +0100
Subject: Add new device idVendor=04b4, idProduct=fd15

Backport commit bdd3c5006c07a801dd2e29162368ac407ac756b4 from upstream,
picking only the required bits.
--- a/src/sispm_ctl.c
+++ b/src/sispm_ctl.c
@@ -164,7 +164,9 @@ int check_outlet_number(int id, int outl
         fprintf(stderr,"mSIS-PM devices only feature one outlet. Number changed from %d to 1\n", outlet);
     outlet = 1;
   }
-  if (id == PRODUCT_ID_SISPM || id == PRODUCT_ID_SISPM_FLASH_NEW) {
+  if ((id == PRODUCT_ID_SISPM) ||
+      (id == PRODUCT_ID_SISPM_FLASH_NEW) ||
+      (id == PRODUCT_ID_SISPM_EG_PMS2)) {
     if (outlet < 1 || outlet > 4) {
       if (verbose == 1)
         fprintf(stderr,"SIS-PM devices only feature 4 outlets. Number changed from %d to 1\n", outlet);
--- a/src/main.c
+++ b/src/main.c
@@ -490,7 +490,9 @@ void parse_command_line(int argc, char*
               printf("%d %s %s\n", status,
                      dev[status]->bus->dirname, dev[status]->filename);
             id = get_id(dev[status]);
-            if ((id == PRODUCT_ID_SISPM) || (id == PRODUCT_ID_SISPM_FLASH_NEW))
+            if ((id == PRODUCT_ID_SISPM) ||
+                (id == PRODUCT_ID_SISPM_FLASH_NEW) ||
+                (id == PRODUCT_ID_SISPM_EG_PMS2))
               if (numeric == 0)
                 printf("device type:      4-socket SiS-PM\n");
               else
@@ -789,7 +791,8 @@ int main(int argc, char** argv)
       if ((dev->descriptor.idVendor == VENDOR_ID) && ((dev->descriptor.idProduct == PRODUCT_ID_SISPM) ||
                                                       (dev->descriptor.idProduct == PRODUCT_ID_MSISPM_OLD) ||
                                                       (dev->descriptor.idProduct == PRODUCT_ID_MSISPM_FLASH) ||
-                                                      (dev->descriptor.idProduct == PRODUCT_ID_SISPM_FLASH_NEW))) {
+                                                      (dev->descriptor.idProduct == PRODUCT_ID_SISPM_FLASH_NEW) ||
+                                                      (dev->descriptor.idProduct == PRODUCT_ID_SISPM_EG_PMS2))) {
 	usbdev[count++] = dev;
       }
       if (count == MAXGEMBIRD) {
--- a/src/sispm_ctl.h
+++ b/src/sispm_ctl.h
@@ -39,6 +39,7 @@
 #define PRODUCT_ID_MSISPM_OLD		 0xFD10
 #define PRODUCT_ID_MSISPM_FLASH		 0xFD12
 #define PRODUCT_ID_SISPM_FLASH_NEW		 0xFD13
+#define PRODUCT_ID_SISPM_EG_PMS2	 0xFD15
 
 
 
